src/java/org/apache/cassandra/hints/HintsStore.java

```diff
@@ -69,6 +73,9 @@ private HintsStore(UUID hostId, File hintsDirectory, ImmutableMap<String, Object
 
         //noinspection resource
         lastUsedTimestamp = descriptors.stream().mapToLong(d -> d.timestamp).max().orElse(0L);
+
+        long hintsNum = descriptors.stream().mapToLong(d -> d.statistics().totalCount()).sum();
+        HintsServiceMetrics.hintsOnDisk.inc(hintsNum);
```

Comment:
Just noticed that we update Counter with existing hint descriptors. I think there no need to do this since we already recorded count for these hints ( counter handles well resets with rate functions ).

Reply:
I see there is not Gauge in C* metrics so I guess we are (mis)using Counter like Gauge. In that regard we can keep this as it is.